packages/block/src/block.ts

```diff
@@ -91,7 +91,7 @@ export class Block {
     const transactions = []
     for (const txData of txsData || []) {
       transactions.push(
-        Transaction.fromValuesArray(txData, {
+        TransactionFactory.fromBlockBodyData(txData, {
```

Comment:
Was thinking a bit about this breaking part in `Block`.

So if I get this right the *only* breaking thing here would be `Transaction` now being a type in the main constructor, right?

Wouldn't we already back on the compatibility track if we rename this new type `Transaction` -> `TypedTransaction` (not sure, or another name, I would think though that this name is justified even if it encompassed the old transaction format, to be discussed) and then `LegacyTransaction` back to `Transaction`?

I actually like this in general (we might still want to rename back `Transaction` -> `LegacyTransaction` on a general breaking-release-round on all libraries, not sure), since having the type named `TypedTransaction` more clearly indicates that a function now allows for a typed tx as an input.

Reply:
We can do this, but this still implies we have to use the `TransactionFactory` in `Block`, and we also have to use the `TypedTransactions` as the type for the `transactions` field in the block body. We can indeed export the `LegacyTransaction` as a `Transaction`, this would mean a non-breaking release for `Transaction`, but due to the type changes on `Block` they are breaking. (I am not entirely sure if they are completely breaking, since all methods of the old `Transaction` are available on both `EIP2930Transaction` and `LegacyTransaction`, so maybe it is not strictly breaking?)

Reply:
The `TransactionFactory` in `Block` is not a problem. With a new `Block` release the `Transaction` dependency is also bumped, so the new `Block` version *for sure* works together with the new `Transaction` version release. So that is no problem.

Compatibility problems only arise on direct API usage, there, the only thing we have is the constructor:

```typescript
constructor(
    header?: BlockHeader,
    transactions: Transaction[] = [],
    uncleHeaders: BlockHeader[] = [],
    opts: BlockOptions = {}
  ) 
```

This is the only part which we need to make sure that it still takes in `Transaction` objects from the current library version, which I think it should when we have a type `EIP2930Transaction | Transaction` (we should test this though, this would be an application for @cgewecke using the functionality from #1118.

If there is the possibility we should very much do this. This is so so crucial that we remain backwards-compatible (especially if it's just a case about a simple rename).

Reply:
I will rebase #1118 against this an open as draft.

Reply:
Cool! 😄  Can you do the suggested renaming (see above) also along for test purposes? So this would be needed to see if this will work.

Any opinion if we now might want to merge this in here? I would be still a fan, think this would ease follow-up development. Doesn't mean that we take this out of review. We can just do follow-up fixes on future PRs.